pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -123,6 +122,41 @@ func (p *adProvider) getPrincipalsFromSearchResult(result *ldapv2.SearchResult,
 	}
 	userPrincipal.Me = true
 
+	if config.NestedGroupMembershipEnabled != nil {
+		if *config.NestedGroupMembershipEnabled == false {
```

Comment:
Is there any reason why we are referring to the flag in different ways here? 
config.NestedGroupMembershipEnabled and *config.NestedGroupMembershipEnabled?

Reply:
The first to check if the pointer is not nil